src/main/java/org/jenkinsci/plugins/oic/OicPropertyDescriptor.java

```diff
@@ -0,0 +1,28 @@
+package org.jenkinsci.plugins.oic;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.Descriptor;
+import org.pac4j.oidc.config.OidcConfiguration;
+
+public abstract class OicPropertyDescriptor extends Descriptor<OicProperty> implements ExtensionPoint {
+    public static ExtensionList<OicPropertyDescriptor> all() {
+        return ExtensionList.lookup(OicPropertyDescriptor.class);
+    }
+
+    /**
+     * Allows the property to restrict its applicability depending on the context (for example, FIPS)
```

Comment:
When I first saw this I was confused, perhaps some more docs on why this is here and when to use it vs not registering your Descriptor would help?  e.g.

```suggestion
     * Allows the property to restrict its applicability depending on the context (for example, FIPS).
     * This is different to not registering the {@code Descriptor} in Jenkins so that default (fallback) configuration can be used.
     * @see #getFallbackConfiguration
```